citesphere/src/main/java/edu/asu/diging/citesphere/web/user/authorities/AuthorityEntryController.java

```diff
@@ -51,15 +55,47 @@
             @RequestParam(defaultValue = "10", required = false, value = "pageSize") int pageSize,
             @RequestParam("firstName") String firstName, @RequestParam("lastName") String lastName) {
 
-        AuthoritySearchResult authorityResult = new AuthoritySearchResult();
-
         List<IAuthorityEntry> userEntries = authorityService.findByName((IUser) authentication.getPrincipal(),
-                firstName, lastName, page, pageSize);
+                firstName, lastName);
+
+        if (!firstName.trim().isEmpty() && !lastName.trim().isEmpty()) {
+            Set<String> userEntryIds = userEntries.stream().map(IAuthorityEntry::getId).collect(Collectors.toSet());
+            List<IAuthorityEntry> lastNameEntries = authorityService
+                    .findByName((IUser) authentication.getPrincipal(), "", lastName).stream()
+                    .filter(lastNameEntry -> !userEntryIds.contains(lastNameEntry.getId()))
+                    .collect(Collectors.toList());
+            userEntries.addAll(lastNameEntries);
+        }
```

Comment:
this doesn't seem like a good approach. If there are 1000 results, then all 1000 results will be loaded into memory. The pagination needs to stay. I would just after you retrieved the page for first/last name search, check if there are fewer results than page size and if so fill in the rest with the ""/last name search. You can keep track of the pages for ""/last name search by looking at the past page (and how big the difference to total pages of first/last name search is).